### ISSUE TITLE
Android Release Build issue

### ISSUE THREAD

**Issue by andgarfi** [sonikro/flutter-okta-sdk#7]
Hey there

I've been using your library for almost 3-4 weeks and everything was working perfectly, the login/logout flow works perfectly during development, then I created a android apk release build for for testing and I noticed the webview flow stops working, the web browser don't open and I get a Platform exception, the strange part is everything works in a release mode. Have you tried to create a release build before for your sample app?

Regards and thanks!

**Comment by Perdiga** [comment]
Hi @andgarfi, we didn´t publish any application using this code, we just tested it on our cellphones and emulators.
Could you please share the stack trace with us, maybe we can help you

**Comment by andgarfi** [comment]
Hey @Perdiga 

This is only error I see when running the release build

2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)
2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)

Problem I have is everything works on debug mode build so there us not an easy way to replicate this, the only way will be create a release build and try it.

Thanks !

**Comment by Perdiga** [comment]
Did you enabled `Authorization Code` and `Refresh Token`

**Comment by Perdiga** [comment]
> Hey @Perdiga
> 
> This is only error I see when running the release build
> 
> 2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)
> 2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)
> 
> Problem I have is everything works on debug mode build so there us not an easy way to replicate this, the only way will be create a release build and try it.
> 
> Thanks !

Can you share a repo where I can reproduce the error ? The error that you are seeing is triggered on FlutterOktaSdkPligin.kt line 127, basically is a generic error 

Which method are you calling ?

**Comment by andgarfi** [comment]
Hey @Perdiga I was able to reproduce this error using the sample app(https://github.com/Perdiga/flutter_okta_sample) I did the steps recommended on the flutter site https://flutter.dev/docs/deployment/android

Then I generated the APK file by running the command flutter build apk and the flutter install

If you do that the app will be installed in your phone in release mode and you will se the error, in this case the Authorize button on the initial screen will NOT open the webview and triggers that errors

2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)
2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)

**Comment by Perdiga** [comment]
> Hey @Perdiga I was able to reproduce this error using the sample app(https://github.com/Perdiga/flutter_okta_sample) I did the steps recommended on the flutter site https://flutter.dev/docs/deployment/android
> 
> Then I generated the APK file by running the command flutter build apk and the flutter install
> 
> If you do that the app will be installed in your phone in release mode and you will se the error, in this case the Authorize button on the initial screen will NOT open the webview and triggers that errors
> 
> 2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)
> 2020-12-21 16:22:29.339 15024-15056/? I/flutter: PlatformException(-1100, tokens must not be null, null, null)

Hey @andgarfi good news, I was able to reproduce your error. I fixed it by generating a release key as demonstrated here
https://flutter.dev/docs/deployment/android

Did you created this key before running `flutter run --release`? 

By generating this key I was able to run the code on an emulator and real device without problems in release mode

I will find a way to throw an error so this issue can be easily tracked.

Please let me know if this fix worked for you

**Comment by andgarfi** [comment]
Hey @Perdiga 

So, I did all the steps to create the keystore many times, as the flutter documentation describes, and If I run 'flutter run --release' everything works just fine but If I generate/build an apk  or an appbundle and then try to install those in a device for a final test, then that's when I get the error

I'm running the following commands:
for APK: flutter build apk
for appbundle and then extract the apks file: flutter build appbundle

Are you able to do this?

Thanks!

**Comment by Perdiga** [comment]
I was able to reproduce, but not able to fix it, I created an issue on flutter github since there were some people with similar issues.
I will keep trying here

**Comment by andgarfi** [comment]
@Perdiga Thank you very much for your help with this!, I believe the flutter run --release could make the trick for now.

**Comment by dnamicandresg** [comment]
@Perdiga 

Remember this issue? I finally solved, the problem was the way android generates the package for the okta SDK, I debugged your flutter package and the real error was coming from the OKTA SDK and says "Invalid discovery document"

Is exactly this: https://github.com/okta/okta-oidc-android/issues/132

The solution is 

1. Create a proguard-rules.pro file inside the android/app folder and add the following rule

-ignorewarnings
-keep class com.okta.oidc.** { *; }

2. Add a couple of rules to the buildTypes/release block inside the app/build.gradle file

   buildTypes {
        release {
            useProguard true
            proguardFiles getDefaultProguardFile('proguard-android.txt'),
                    'proguard-rules.pro'
            signingConfig signingConfigs.release
        }
    }


I believe this should be included inside the README file of the REPO as part of the setup process

Hope it helps!
